docs/src/main/asciidoc/quarkus-virtual-threads.adoc

```diff
@@ -0,0 +1,490 @@
+
+=== Differences between virtual threads and platform threads
+We will give a brief overview of the topic here, please refer to the link:{vthreadjep}[JEP 425] for more information.
+
+Virtual threads are a feature available since Java 19 aiming at providing a cheap alternative to platform threads for
+IO-bound workloads.
+
+Until now, platform threads were the concurrency unit of the JVM. They are a thin wrapper over OS structures.
+This means that creating a Java platform thread actually results in creating a "thread-like" structure in your operating
+system.
+
+Virtual threads on the other hand are managed by the JVM. In order to be executed, they need to be mounted on a platform
+thread (which acts as a carrier).
+As such, they have been designed to offer the following characteristics:
+
+Lightweight :: Virtual threads occupy less space than platform threads in memory.
+Hence, it becomes possible to use more virtual threads than platform threads simultaneously without blowing up the heap.
+By default, platform threads are created with a stack of about 1 MB (need ref) where virtual threads stack is
+"pay-as-you-go".
+
+Cheap to create:: Creating a platform thread in Java takes time.
+Currently, techniques such as pooling where threads are created once then reused are strongly encouraged to minimize the
+time lost in starting them (as well as limiting the maximum number of threads to keep memory consumption low).
+Those techniques are now outdated: virtual threads are supposed to be disposable entities that we create when we need
+them, it is discouraged to pool them or to reuse them for different tasks.
+
+Cheap to block:: When performing blocking IO, the underlying OS thread wrapped by the Java platform thread is put in a
+wait queue and a context switch occurs to load a new thread context onto the CPU core. This operation takes time. Since
+virtual threads are managed by the JVM, no underlying OS thread is blocked when they perform a blocking operation. Their
+state is simply stored in the heap and another Virtual thread is executed on the same Java platform thread.
+
+=== Virtual threads are useful for IO-bound workloads only
+We now know that we can create way more virtual threads than platform threads. One could be tempted to use virtual
+threads to perform long computations (CPU-bound workload). This is useless if not counterproductive.
+CPU-bound doesn't consist in quickly swapping threads while they need to wait for the completion of an IO but in leaving
+them attached to a CPU-core to actually compute something. In this scenario, it is useless to have thousands of threads
+if we have tens of CPU-cores, virtual threads won't enhance the performance of CPU-bound workloads.
+
+
+== Bringing virtual threads to reactive REST services
+Since virtual threads are disposable entities, the fundamental idea of quarkus-loom is to offload the execution of an
```

Comment:
we need a new name. quarkus-loom is not great.
What about the Quarkus Virtual Thread feature or support?

Reply:
go with it.